addons/knowledge/models/knowledge_article.py

```diff
@@ -599,16 +599,20 @@ def create(self, vals_list):
     def write(self, vals):
         # Move under a parent is considered as a write on it (permissions, ...)
         _resequence = False
-        if vals.get('parent_id'):
+        if 'parent_id' in vals:
             parent = self.browse(vals['parent_id'])
-            try:
-                parent.check_access_rights('write')
-                parent.check_access_rule('write')
-            except AccessError:
-                raise AccessError(_("You cannot move an article under %(parent_name)s as you cannot write on it",
-                                    parent_name=parent.display_name))
+            if parent:
+                try:
+                    parent.check_access_rights('write')
+                    parent.check_access_rule('write')
+                except AccessError:
+                    raise AccessError(_("You cannot move an article under %(parent_name)s as you cannot write on it",
+                                        parent_name=parent.display_name))
             if 'sequence' not in vals:
-                max_sequence = self._get_max_sequence_inside_parents(parent.ids).get(parent.id, -1)
+                if parent:
+                    max_sequence = self._get_max_sequence_inside_parents(parent.ids).get(parent.id, -1)
+                else:
+                    max_sequence = self._get_max_sequence_inside_parents(False).get(False, -1)
```

Comment:
As you modified _get_max_sequence_inside_parents to handle empty parent_ids, you can just do
`max_sequence = self._get_max_sequence_inside_parents(parent.ids).get(parent.id, -1)`
it will work for both cases